README.rst

```diff
@@ -26,6 +26,62 @@ What is OOP Extensions ?
 
 OOP Extensions is a set of utilities for object oriented programming which is missing on Python core libraries.
 
+Usage
+================================================================================
+``oop_ext`` brings a set of object oriented utilities, it supports the concept of interfaces,
+abstract/overridable methods and more. ``oop_ext`` carefully checks that implementations
+have the same method signatures as the interface it implements and raises exceptions otherwise.
+
+Here's a simple example showing some nice features:
+
+.. code-block:: python
+
+    from oop_ext.foundation.decorators import Override, Abstract
+    from oop_ext.interface import Interface, ImplementsInterface, IsImplementation
+
+    class IDisposable(Interface):
+        def dispose(self):
+            """
+            Clears this object
+            """ 
+        def is_disposed(self) -> bool:
+            """
+            Returns True if the object has been cleared
+            """
+
+    @ImplementsInterface(IDisposable)
+    class Disposable:
```

Comment:
I wouldn't mix `Abstract` and `interfaces` in the same example as it might give the impression that you *need* to use both of them at the same time, but I will leave that up to you. 😁

Reply:
Good call, I'll leave only one `Abstract` and `Override` both methods.